package.json

```
@@ -22,6 +22,7 @@
     "build:ssr": "cd ssr && webpack --mode=production",
     "build:sw": "cd client/pwa && yarn && yarn build:prod",
     "build:sw-dev": "cd client/pwa && yarn && yarn build",
+    "clean": "git clean -dix -e '.env'",
```

@LeoMcA How about two commands:

```suggestion
    "clean": "git clean -i '**/*.js' '**/*.js.map'",
    "clean:all": "git clean -dix -e '.env'",
```

---

Ah, as soon as those files are in `.gitignore`, `git clean -i` doesn't catch it. `git clean -iX` would, but it ignores additional parameters, so we cannot limit it to `**/*.js` and `**/.js.map`. 🤔

---

I'm sortof not entirely sure the transpiled .js files *should* be in `.gitignore` though - not sure it's entirely logical, maybe more of a gut feeling, but imo files which break stuff (as these .js files do) shouldn't be hidden. As in "oh no the build isn't working let's run `git status` to see what's wrong, *ah* that explains it".

This is how I removed these files before, because they weren't in `.gitignore`.

---

On the one hand, I tend to agree, but on the other hand, there are files e.g. in `client/build/*` and `ssr/dist` that we would keep in `.gitignore` yet those are build artifacts that we might want to cleanup with a `clean` command. Wdyt? 🤔

---

Indeed, though I would clean those with the more broad command, as the mere existence of those shouldn't break anything. As I see it, one clean should be for "I filled my environment with transpiled files undo that please and put it back how it was" the other should be "I have no idea why my environment is broken, delete everything please and let me start afresh".